typegen/strats/inline.py

```diff
@@ -101,67 +103,104 @@ def visit_FunctionDef(self, fdef: ast.FunctionDef):
+        for target_name_with_node in target_names_with_nodes:
+            target_name = target_name_with_node[0]
+            target_node = target_name_with_node[1]
+            target_trace_data = None
+            if isinstance(target_node, ast.Attribute):
+                # Finds the class member type hint.
+                target_trace_data = relevant_trace_data_class_member[
+                    relevant_trace_data_class_member[TraceData.VARNAME] == target_name]
+            elif isinstance(target_node, ast.Name):
+                # Finds the local variable type hint.
+                target_trace_data = relevant_trace_data_local_variables[
+                    relevant_trace_data_local_variables[TraceData.VARNAME] == target_name]
+            if target_trace_data is None or len(target_trace_data) == 0:
+                logger.debug(
+                    f"No hint found for assign for '{target_name}'"
+                )
+                continue
+
             logger.debug(
-                f"Applying type hints for simple assignment '{node.targets[0].id}'"
-            )
-            return ast.AnnAssign(
-                target=node.targets[0],
-                value=node.value,
-                annotation=ast.Name(node_vars[TraceData.VARTYPE].values[0]),
-                simple=True,
+                f"Applying type hints for simple assignment '{target_name}'"
             )
-
-        else:
-            prehints = list(
-                node_vars[[TraceData.VARNAME, TraceData.VARTYPE]].itertuples(
-                    index=False, name=None
+            if contains_one_target:
+                new_node = ast.AnnAssign(
+                    target_node,
+                    value=node.value,
+                    annotation=ast.Name(target_trace_data[TraceData.VARTYPE].values[0]),
+                    simple=True,
                 )
-            )
-            logger.debug(f"Applying type hints for multi-assignments '{prehints}'")
-
-            assigns = [
-                ast.AnnAssign(
-                    target=ast.Name(name), annotation=ast.Name(hint), simple=True
+                return new_node
+            else:
+                new_node = ast.AnnAssign(
+                    target_node,
+                    annotation=ast.Name(target_trace_data[TraceData.VARTYPE].values[0]),
+                    simple=True
                 )
-                for name, hint in prehints
-            ]
 
-            newhints: list[ast.AST] = list()
-            newhints.extend(assigns)
-            newhints.append(node)
+            new_nodes.append(new_node)
 
-            return newhints
+        new_nodes.append(node)
+        return new_nodes
 
-    def _extract_assign_ids(self, node: ast.AST) -> list[str]:
-        # https://stackoverflow.com/a/72231602
-        return [i.id for i in ast.walk(node) if isinstance(i, ast.Name)]
+    def _extract_target_names_with_nodes(self, node: ast.AST) -> list[typing.Tuple[str, ast.Name | ast.Attribute]]:
```

Comment:
technically correct, but prefer to use `tuple` instead of `typing.Tuple`

Reply:
Fixed